### ISSUE TITLE
Train test split overflow

### ISSUE THREAD

**Issue by vindex10** [Accenture/AmpliGraph#90]
### Description
`train_test_split_no_seen()` function always fails without `allow_duplicates=True`. Tests pass by accident.

### Actual Behavior
```
Exception: Not possible to split the dataset.                                 Maybe set allow_duplication = True can help...
```

### Expected Behavior
split dataset in the output

### Steps to Reproduce
Try changing test set into the following (in the `tests/ampligraph/evaluation/protocol.py`):
```
    X = np.array([['a', 'y', 'b'],
                  ['x', 'y', 'c'], # changed 'a' to 'x' at the 0 position
                  ['c', 'y', 'a'],
                  ['d', 'y', 'e'],
                  ['e', 'y', 'f'],
                  ['f', 'y', 'c'],
                  ['f', 'y', 'c']])

    expected_X_train = np.array([['a', 'y', 'b'],
                                ['c', 'y', 'a'],
                                ['d', 'y', 'e'],
                                ['e', 'y', 'f'],
                                ['f', 'y', 'c']])

    expected_X_test = np.array([['x', 'y', 'c'], # changed 'a' to 'x' at the 0 position
                                ['f', 'y', 'c']])

```

**Comment by vindex10** [comment]
this condition is required to split dataset while having all entities in the test_set with defined IDs

**Comment by chanlevan** [comment]
Hi @vindex10 

There are some cases not possible to split the train/test datasets without duplication in the test set. As you see from our exception message, the _allow_duplicates_ parameter maybe can help.
